### PR TITLE
Optimize labels.String()

### PR DESCRIPTION
**Pull request by howardjohn**
```
name                 old time/op    new time/op    delta
LabelString/small-8     481ns ± 4%     169ns ± 1%  -64.96%  (p=0.008 n=5+5)
LabelString/big-8      12.2µs ± 3%     4.9µs ± 3%  -59.29%  (p=0.008 n=5+5)

name                 old alloc/op   new alloc/op   delta
LabelString/small-8      216B ± 0%       56B ± 0%  -74.07%  (p=0.008 n=5+5)
LabelString/big-8      6.01kB ± 0%    2.30kB ± 0%  -61.67%  (p=0.008 n=5+5)

name                 old allocs/op  new allocs/op  delta
LabelString/small-8      10.0 ± 0%       2.0 ± 0%  -80.00%  (p=0.008 n=5+5)
LabelString/big-8         158 ± 0%         3 ± 0%  -98.10%  (p=0.008 n=5+5)
```

This is shown to be a hot path in some scenarios (https://github.com/istio/istio/issues/44985)

### CODE REVIEW

File: pkg/config/labels/instance.go [hunk 1]
```diff
@@ -140,25 +140,25 @@ func validateTagKey(k string) error {
...
-			labels = append(labels, k)
-		}
-	}
-	sort.Strings(labels)
+	// Ensure stable ordering
+	keys := slices.Sort(maps.Keys(i))
 
-	var buffer bytes.Buffer
+	var buffer strings.Builder
+	// Assume each kv pair is roughly 10 characters. We could be under or over, this is just a guess to optimize
```
Comment (hzxuzhonghu) [inline pkg/config/labels/instance.go:147]: I am not sure this assumption is right, in real many software has their long domain suffix added to the key

like `app.kubernetes.io/name`:
`cluster.x-k8s.io/provider:`

Comment (howardjohn) [reply pkg/config/labels/instance.go:147]: I agree, figured it was better to underestimate though?

Comment (howardjohn) [reply pkg/config/labels/instance.go]: updated it

**Comment by howardjohn** [comment]
This is less important since https://github.com/istio/istio/pull/45230 removes it from the hotpath. But might as well have it more efficient

**Comment by howardjohn** [comment]
ping @hzxuzhonghu

**Review by hzxuzhonghu — APPROVED** [review]

**Pull request merged**